tests/scanner/audit/external_project_access_engine_test.py

```diff
@@ -242,12 +326,55 @@ def test_single_item_no_match(self):
         self.assertEqual('projects/123,folders/456,organizations/7890',
                          violation.resource_data)
 
+    def test_single_item_no_match_with_user(self):
+        """Test violations are found with single item in rule"""
+        processed_rule = {
+            'ancestor_resources': [Organization('789')],
+            'users': ['user1@example.com']
+        }
+
+        rule = engine_module.Rule(rule_name='test_single_item_no_match_with_user',
+                                  rule_index=0,
+                                  rules=processed_rule)
+        violation = rule.find_violation('user1@example.com',
+                                        self.TEST_ANCESTORS)
+
+        self.assertEqual(0, violation.rule_index)
+        self.assertEqual('test_single_item_no_match_with_user',
+                         violation.rule_name)
+        self.assertEqual('projects/123', violation.full_name)
+        self.assertEqual('projects/123,folders/456,organizations/7890',
+                         violation.resource_data)
+
+    def test_single_item_no_match_with_wrong_user(self):
+        """Test violations are found with single item in rule"""
```

Comment:
ditto on description, add "with wrong user"; can you please update other descriptions too?